pkg/internal/controllers/certificaterequests.go

```diff
@@ -195,8 +200,8 @@ func (c *certificaterequests) reconcileStatusPatch(ctx context.Context, req ctrl
 		log.V(2).Info("approving request")
 		c.recorder.Event(cr, corev1.EventTypeNormal, "Approved", response.Message)
 
-		c.setCertificateRequestStatusCondition(
-			&crPatch.Conditions,
+		crPatch.Conditions = c.getCertificateRequestStatusConditions(
```

Comment:
@erikgb do we still need these changes?

Reply:
I think the lines 179-183 do fix the issue already right?

Reply:
Yes, either will fix issues. This is a bug IMO - regardless of the early exit.

Reply:
The early exit just hides the bug. But if the cache is stale, we will be hit if not fixing this also.

Reply:
I also noticed this now "duplicated" line:

https://github.com/cert-manager/approver-policy/blob/c8f950602e17a0a6fc0d0ef78dd61db20b387d59/pkg/internal/controllers/certificaterequests.go#L100

This should probably be cleaned up. And makes me even more sure we must have the `lastTransitionTime` bugfix.

Reply:
I suggest we now remove the duplicated early exit, ref.

https://github.com/cert-manager/approver-policy/blob/c8f950602e17a0a6fc0d0ef78dd61db20b387d59/pkg/internal/controllers/certificaterequests.go#L115-L118

Isn't a predicate better and sufficient?